fieldservice_sale/models/sale_order.py

```diff
@@ -103,7 +103,15 @@ def _field_service_generate_sale_fsm_orders(self, new_fsm_sol):
 
         if new_fsm_sol:
             fsm_by_sale = self.env["fsm.order"].search(
-                [("sale_id", "=", self.id), ("sale_line_id", "=", False)]
+                [
+                    ("sale_id", "=", self.id),
+                    ("sale_line_id", "=", False),
+                    (
+                        "stage_id",
+                        "!=",
+                        self.env.ref("fieldservice.fsm_stage_cancelled").id,
```

Comment:
@ppyczko I am pretty sure this will not work in a multi-company environment. You should use `is_closed` on the FSM order:

https://github.com/APSL/field-service/blob/15.0-imp-fieldservice_sale-cancel-order/fieldservice/models/fsm_order.py#L77

Reply:
Thank you for your feedback @max3903! I've applied the requested changes, let me know if there’s anything else to adjust!